lock-impl/src/main/java/com/palantir/lock/impl/LockServerSync.java

```diff
@@ -267,4 +268,12 @@ private synchronized void decrementReadCount(int clientIndex) {
         return Collections.unmodifiableList(
                 readLockHolders.keysView().collect(clients::fromIndex, new ArrayList<>(readLockHolders.size())));
     }
+
+    synchronized Optional<LockClient> getWriteClient() {
+        if (getState() > 0) {
+            return Optional.of(clients.fromIndex(writeLockHolder));
+        } else {
+            return Optional.empty();
+        }
```

Comment:
extreme nit-pickyness: you'll notice that the style here is generally not to have the else clause after a return, so that should be followed (I know this is different in AtlasDB Proxy)